File: examples/simple/ThriftProducer.cpp
#include "../../lib/cpp/producer.h"
#include <example_types.h>

int main(int argc, char** argv)
{
	jp::ThriftProducer<ExampleData> p("thrift");

	ExampleData doc;
	doc.language = "C++";
	doc.api = "simple";
	doc.format = "Thrift";

	p.add(doc);
	return 0;
}

File: examples/simple/ThriftProducer.rb
#!/usr/bin/env ruby
$LOAD_PATH.push '../../lib/rb'
$LOAD_PATH.push '../gen-rb/'
require 'producer'
require 'example_types'

p = Jp::ThriftProducer.new 'thrift'
doc = ExampleData.new language: 'ruby', api: 'simple', format: 'thrift'
p.add doc
